src/app/src/main/kotlin/org/imperial/mrc/hint/security/oauth2/ProfileDefinition.kt

```diff
@@ -37,9 +33,6 @@ class ProfileDefinition(
 
     override fun getProfileUrl(accessToken: Token?, configuration: OAuthConfiguration?): String
     {
-        accessToken.let {
-           token = (accessToken as OAuth2AccessToken).accessToken
-        }
         return "https://${appProperties.oauth2ClientUrl}/userinfo"
     }
```

Comment:
Why was this change necessary, and how does it affect behaviour?

Reply:
This was never the right way to retrieve access token. This PR has updated the code to retrieve token from the session. Pac4j OAuth20 stores access and refresh token in sessions.